Wrappers/d3d8/Overlay.cpp

```diff
@@ -155,118 +177,129 @@ void Overlay::DrawDebugOverlay(LPDIRECT3DDEVICE8 ProxyInterface)
-	OvlString = "Room ID: 0x";
+	OvlString.append("\rRoom ID: 0x");
 	OvlString.append(IntToHexStr(GetRoomID()));
 
-	TextStruct.String = OvlString.c_str();
-	TextStruct.Rect.top += TextSpacing;
-	TextStruct.Rect.bottom += TextSpacing;
-
-	DrawDebugText(ProxyInterface, TextStruct);
-
-	OvlString = "Cutscene ID: 0x";
+	OvlString.append("\rCutscene ID: 0x");
 	OvlString.append(IntToHexStr(GetCutsceneID()));
 
-	TextStruct.String = OvlString.c_str();
-	TextStruct.Rect.top += TextSpacing;
-	TextStruct.Rect.bottom += TextSpacing;
+	OvlString.append("\rFPS: ");
+	OvlString.append(FloatToStr(GetFPSCounter(), FloatPrecision));
 
-	DrawDebugText(ProxyInterface, TextStruct);
+	OvlString.append("\rChar X Position: ");
+	OvlString.append(FloatToStr(GetJamesPosX(), FloatPrecision));
 
-	OvlString = "FPS: ";
-	OvlString.append(FloatToStr(GetFPSCounter(), FloatPrecision));
+	OvlString.append("\rChar Y Position: ");
+	OvlString.append(FloatToStr(CharYPos, FloatPrecision));
 
-	TextStruct.String = OvlString.c_str();
-	TextStruct.Rect.top += TextSpacing;
-	TextStruct.Rect.bottom += TextSpacing;
+	OvlString.append("\rChar Z Position: ");
+	OvlString.append(FloatToStr(GetJamesPosZ(), FloatPrecision));
 
-	DrawDebugText(ProxyInterface, TextStruct);
+	DebugOverlayTextStruct.String = OvlString.c_str();
+	DrawDebugText(ProxyInterface, DebugOverlayTextStruct);
 
-	OvlString = "Char X Position: ";
-	OvlString.append(FloatToStr(GetJamesPosX(), FloatPrecision));
+}
 
-	TextStruct.String = OvlString.c_str();
-	TextStruct.Rect.top += TextSpacing;
-	TextStruct.Rect.bottom += TextSpacing;
+void Overlay::DrawDebugText(LPDIRECT3DDEVICE8 ProxyInterface, Overlay::D3D8TEXT FontStruct)
+{
+	Logging::LogDebug() << __FUNCTION__;
+	
+	RECT DropShadowRect = FontStruct.Rect;
+	DropShadowRect.top = DropShadowRect.top + DropShadowOffset;
+	DropShadowRect.left = DropShadowRect.left + DropShadowOffset;
 
-	DrawDebugText(ProxyInterface, TextStruct);
+	if (ResetDebugFontFlag)
+	{
+		ResetDebugFontFlag = false;
+		DebugFont->OnResetDevice();
+	}
 
-	OvlString = "Char Y Position: ";
-	OvlString.append(FloatToStr(CharYPos, FloatPrecision));
+	if (ProxyInterface != nullptr && DebugFont == nullptr)
+	{
+		HFONT FontCharacteristics = CreateFontA(16, 0, 0, 0, FW_BOLD, 0, 0, 0, 0, 0, 0, PROOF_QUALITY, 0, FontName);
+		if (FontCharacteristics != NULL)
+		{
+			Logging::LogDebug() << __FUNCTION__ << " Creating Debug font: " << FontName;
+			D3DXCreateFont(ProxyInterface, FontCharacteristics, &DebugFont);
+			DeleteObject(FontCharacteristics);
+		}
+	}
+
+	if (DebugFont != nullptr)
+	{
+		DebugFont->DrawTextA(FontStruct.String, -1, &DropShadowRect, FontStruct.Format, TextColors.Black);
+		DebugFont->DrawTextA(FontStruct.String, -1, &FontStruct.Rect, FontStruct.Format, FontStruct.Color);
+	}
+}
 
-	TextStruct.String = OvlString.c_str();
-	TextStruct.Rect.top += TextSpacing;
-	TextStruct.Rect.bottom += TextSpacing;
+void Overlay::DrawMenuTestText(LPDIRECT3DDEVICE8 ProxyInterface, Overlay::D3D8TEXT FontStruct)
```

Comment:
Can you rename this function `DrawMenuTestText()` to something more descriptive?  Maybe just remove the word "Test" from it?

Reply:
Of course!